mindmeld/augmentation.py

```diff
@@ -0,0 +1,285 @@
+
+    def _read_path_queries(self, path):
+        """Returns all queries in a specified file path.
+
+        Args:
+            path (str): File path.
+
+        Return:
+            queries (list): List of queries in the file path.
+        """
+        with open(path, "r") as f:
+            queries = f.readlines()
+        return queries
+
+    def _write_files(self, path, augmented_queries):
+        """Writes augmented queries to a new file in the path.
+
+        Args:
+            path (str): File path to the original file.
+            augmented_queries (list): List of augmented queries returned by class.
+        """
+        write_path = path.strip(".txt") + ".augmented.txt"
+
+        with open(write_path, "w") as outfile:
+            for query in augmented_queries:
+                outfile.write(query.strip("\n") + "\n")
+
+
+class EnglishParaphraser(Augmentor):
+    """Paraphraser class for generating English paraphrases."""
+
+    def __init__(self, app_path, config=None):
+        """Initializes an English paraphraser.
+
+        Args:
+            app_path (str): The location of the MindMeld app
+            config (dict, optional): A config object to use. This will
+                override the config specified by the app's config.py file.
+        """
+        super().__init__(app_path=app_path, config=config)
+
+        model_name = "tuner007/pegasus_paraphrase"
+        torch_device = "cuda" if torch.cuda.is_available() else "cpu"
+        self.tokenizer = PegasusTokenizer.from_pretrained(model_name)
+        self.model = PegasusForConditionalGeneration.from_pretrained(
+            model_name, force_download=False
+        ).to(torch_device)
+
+    def _get_response(self, query, num_return_sequences=10, num_beams=10):
+        """Generates paraphrase responses for given query.
+
+        Args:
+            query (str): An application query.
+            num_return_sequences (int, optional): Maximum number of paraphrases to be generated.
+            num_beams (int, optional):
+
+        Return:
+            paraphrases (list(str)): List of paraphrased queries.
+        """
+        batch = self.tokenizer.prepare_seq2seq_batch(
+            [query], truncation=True, padding="longest", max_length=60
+        )
+        translated = self.model.generate(
+            **batch,
+            max_length=60,
+            num_beams=num_beams,
+            num_return_sequences=num_return_sequences,
+            temperature=1.5,
+        )
+        paraphrases = self.tokenizer.batch_decode(translated, skip_special_tokens=True)
+        return paraphrases
+
+    def augment_queries(self, queries, **kwargs):
+        augmented_queries = []
+        for query in queries:
+            augmented_queries.extend(self._get_response(query, **kwargs))
+        return augmented_queries
+
+
+class MultiLingualParaphraser(Augmentor):
+    """Paraphraser class for generating paraphrases based on language code of the app
```

Comment:
Modify comment to indicate that this only supports romance languages (French, Italian, Spanish, Portuguese, and Romanian) right now.